include/ginkgo/core/solver/lower_trs.hpp

```diff
@@ -196,23 +91,23 @@ class LowerTrs : public EnableLinOp<LowerTrs<ValueType, IndexType>>,
     }
 
     /**
-     * Gets the right hand side of the linear system.
+     * Returns the preconditioner operator used by the solver.
      *
-     * @return the right hand side
+     * @return the preconditioner operator used by the solver
      */
-    std::shared_ptr<const matrix::Dense<ValueType>> get_rhs() const
+    std::shared_ptr<const LinOp> get_preconditioner() const override
     {
-        return b_;
+        return preconditioner_;
     }
 
     /**
-     * Returns the preconditioner operator used by the solver.
+     * Get the triangular solve struct
      *
-     * @return the preconditioner operator used by the solver
+     * @return the trs solve struct
      */
-    std::shared_ptr<const LinOp> get_preconditioner() const override
+    gko::solver::SolveStruct *get_solve_struct() const
```

Comment:
Yes, returning a `shared_ptr` would indeed be better.

I just wanted to add some clarification with regard to the namespaces. I think it makes it a bit easier as otherwise, it would not be so clear.

Reply:
Actually, not I just remove this getter, as it does not really serve any purpose.